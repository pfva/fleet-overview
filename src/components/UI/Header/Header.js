import styles from './Header.module.css';

const Header = () => {
  return (
      <header className={styles.header}>
        <h1 className={styles.title}>Welcome, Fleet Owner</h1>
        <p className={styles.infoText}>Here you can view a full list of your vehicle fleet</p>
      </header>
  );
};

export default Header;
