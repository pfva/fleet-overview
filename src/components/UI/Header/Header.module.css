.header {
  display: flex;
  padding: calc(var(--spacing-extra-large) + var(--spacing-normal)) var(--spacing-normal);
  background: linear-gradient(var(--color-white), 99%, var(--color-grey-one));
}

.textWrapper {
  display: flex;
  flex-direction: column;
  margin-left: var(--spacing-extra-large);
}

@media screen and (max-width: 425px) {
  .textWrapper {
    margin-left: 0;
  }
}

.title {
  margin-top: var(--spacing-small);
  font-size: var(--font-size-title);
}

.infoText {
  color: var(--color-grey-three);
  margin-bottom: var(--spacing-extra-small);
}
