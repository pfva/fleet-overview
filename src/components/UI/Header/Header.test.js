import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('Header', () => {
  it('renders without crashing', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the welcome message', () => {
    render(<Header />);
    const title = screen.getByText(/Welcome, Fleet Owner/i);
    expect(title).toBeInTheDocument();
  });

  it('renders the information text', () => {
    render(<Header />);
    const infoText = screen.getByText(/Here you can view a full list/i);
    expect(infoText).toBeInTheDocument();
  });
});
