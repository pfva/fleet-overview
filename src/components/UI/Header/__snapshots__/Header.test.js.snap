// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
<header
  className="header"
>
  <h1
    className="title"
  >
    Welcome, Fleet Owner
  </h1>
  <p
    className="infoText"
  >
    Here you can view a full list of your vehicle fleet
  </p>
</header>
`;
